source/Server/IntegratedAuthentication/IntegratedAuthenticationHost.cs

```diff
@@ -0,0 +1,135 @@
+                foreach (var baseUri in prefixes)
+                {
+                    var prefix = baseUri.ToString();
+            
+                    if (!baseUri.Host.Contains("."))
+                    {
+                        prefix = prefix.Replace("localhost", "+");
+                    }
+            
+                    options.UrlPrefixes.Add(prefix);
+                }
+            });
+            
+             builder.Configure(app =>
+             {
+                 app.Use((context, func) =>
+                 {
+                     if (string.IsNullOrWhiteSpace(context.User.Identity.Name))
+                     {
+                         context.Response.StatusCode = 401;
+                         return Task.CompletedTask;
+                     }
+            
+                     handler.HandleRequest(context);
+                     return Task.CompletedTask;
+                 });
+             });
+            
+            var host = builder.Build();
+            host.Start();
+        }
+
+        AuthenticationSchemes MapAuthenticationScheme()
+        {
+            switch (configurationStore.GetAuthenticationScheme())
+            {
+                case System.Net.AuthenticationSchemes.Ntlm:
+                    return AuthenticationSchemes.NTLM;
+                case System.Net.AuthenticationSchemes.Negotiate:
+                    return AuthenticationSchemes.Negotiate;
+                case System.Net.AuthenticationSchemes.IntegratedWindowsAuthentication:
+                    return AuthenticationSchemes.Kerberos;
+            }
+            return AuthenticationSchemes.None;
+        }
+
+        Uri[] GetListenPrefixes()
+        {
+            var setting = configuration.GetCurrentNodeWebPortalConfiguration().ListenPrefixes;
+
+            var prefixes = new List<Uri>();
+
+            if (setting.Any())
+            {
+                var values = setting.Select(v => v.Trim()).Where(v => v.Length > 0).ToArray();
+
+                foreach (var value in values)
+                {
+                    log.Trace("Adding listen prefix: " + value);
+
+                    try
+                    {
+                        prefixes.Add(new Uri(value.TrimEnd('/') + DirectoryServicesConstants.IntegratedAuthVirtualDirectory, UriKind.Absolute));
+                    }
+                    catch (Exception ex)
+                    {
+                        log.Error(ex);
+                        throw new UriFormatException("Unable to parse listen prefix '" + value + "': " + ex.Message, ex);
+                    }
+                }
+            }
+
+            if (prefixes.Count == 0)
+            {
+                log.Trace("No HTTP listen prefixes were provided; defaulting to http://localhost:8050" + DirectoryServicesConstants.IntegratedAuthVirtualDirectory);
+                prefixes.Add(new Uri("http://localhost:8050" + DirectoryServicesConstants.IntegratedAuthVirtualDirectory, UriKind.Absolute));
```

Comment:
How does this default actually work? Does something else know about this magical default port?